lib/cloud_controller/dependency_locator.rb

```diff
@@ -432,11 +432,19 @@ def build_task_client
     end
 
     def build_opi_task_client
-      ::OPI::TaskClient.new(config)
+      ::OPI::TaskClient.new(config, build_task_completion_callback_generator, VCAP::CloudController::Diego::TaskEnvironmentVariableCollector)
     end
 
     def build_bbs_task_client
-      VCAP::CloudController::Diego::BbsTaskClient.new(build_bbs_client)
+      VCAP::CloudController::Diego::BbsTaskClient.new(config, build_bbs_client, build_task_recipe_builder)
+    end
+
+    def build_task_recipe_builder
+      VCAP::CloudController::Diego::TaskRecipeBuilder.new
+    end
+
+    def build_task_completion_callback_generator
+      VCAP::CloudController::Diego::TaskCompletionCallbackGenerator.new(config)
```

Comment:
This is something that is used in the Diego workflow as well, and is instantiated inside the method here - https://github.com/cloudfoundry/cloud_controller_ng/blob/7a977015f82c2799c6c84906812bb7fa38c09c92/lib/cloud_controller/diego/task_recipe_builder.rb#L19

I feel like we should pick one style over another (instantiating directly within the class versus dependency injected)